In [11]:
#importing pacakages

import numpy as np

import pandas as pd


In [13]:
data=pd.read_csv("/content/insurance.csv")

In [14]:
data.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [15]:
data.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

In [33]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [16]:
data.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

# ONE HOT ENCODING WHERE WE CONVERT THE CATEGORIAL VARIABLE INTO NUMERICAL VARIABLE

In [17]:
data['sex']=data['sex'].apply({'male':1,'female':0}.get)
data['smoker']=data['smoker'].apply({'yes':1,'no':0}.get)
data['region']=data['region'].apply({'northeast':1,'northwest':2,'southeast':3,'southwest':4}.get)

data.head()

age  sex     bmi  children  smoker  region      charges
0   19    0  27.900         0       1       4  16884.92400
1   18    1  33.770         1       0       3   1725.55230
2   28    1  33.000         3       0       3   4449.46200
3   33    1  22.705         0       0       2  21984.47061
4   32    1  28.880         0       0       2   3866.85520

# DIVIDING THE DATA INTO DEPENDENT COLUMNS AND INDEPENDENT COLUMS.

In [27]:
x=data.iloc[:,:-1]             # INDEPENDENT COLUMNS.

x.head()

age  sex     bmi  children  smoker  region
0   19    0  27.900         0       1       4
1   18    1  33.770         1       0       3
2   28    1  33.000         3       0       3
3   33    1  22.705         0       0       2
4   32    1  28.880         0       0       2

In [28]:
y=data.iloc[:,-1]    # DEPENDENT COLUMN (CHARGES)

y.head()

0    16884.92400
1     1725.55230
2     4449.46200
3    21984.47061
4     3866.85520
Name: charges, dtype: float64

In [20]:
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3)

In [21]:
from sklearn.linear_model import Lasso

model=Lasso(alpha=1)  # alpha is the penalising parameter

model.fit(x_train,y_train)

Lasso(alpha=1, copy_X=True, fit_intercept=True, max_iter=1000, normalize=False,
      positive=False, precompute=False, random_state=None, selection='cyclic',
      tol=0.0001, warm_start=False)

In [40]:
prediction=model.predict(x_test)

# Evaluating the coefficient for the columns

In [22]:
lreg_coefficient=pd.DataFrame()

lreg_coefficient["Columns"]=x_train.columns

lreg_coefficient["Coefficient Estimate"]=pd.Series(model.coef_)

print(lreg_coefficient)

    Columns  Coefficient Estimate
0       age            251.891217
1       sex           -175.523814
2       bmi            348.776295
3  children            498.372377
4    smoker          23423.333877
5    region           -413.648023


# Predicting the insurance cost for the new customer using Lasso Regression

In [23]:
data.head()

age  sex     bmi  children  smoker  region      charges
0   19    0  27.900         0       1       4  16884.92400
1   18    1  33.770         1       0       3   1725.55230
2   28    1  33.000         3       0       3   4449.46200
3   33    1  22.705         0       0       2  21984.47061
4   32    1  28.880         0       0       2   3866.85520

In [24]:
data_new={'age':25,'sex':0,'bmi':32,'children':2,'smoker':1,'region':2}

index=[1]

In [25]:
new_customer_data=pd.DataFrame(data_new,index)

new_customer_data

age  sex  bmi  children  smoker  region
1   25    0   32         2       1       2

In [26]:
prediction_new=model.predict(new_customer_data)

print("The insurance cost for the new customer is ", prediction_new)

The insurance cost for the new customer is  [29427.74111327]


# THERFORE THE NEW CUSTOMER WILL GET INSURANCE OF 29427.7411 WHICH IS PREDICTED BY THE MACHINE LEARNING MODEL USING LASSO REGRESSION.